### PR TITLE
fix: add predicate for harbor cluster controller to resolve reconcile…

### PR DESCRIPTION
**Pull request by chlins**
Fix: https://github.com/goharbor/harbor-operator/issues/303

Signed-off-by: chlins <chlins.zhang@gmail.com>

### CODE REVIEW

File: controllers/goharbor/harborcluster/ctrl_setup.go [hunk 1]
```diff
@@ -90,3 +94,25 @@ func New(ctx context.Context, name string, configStore *configstore.Store) (comm
...
+		if !ok {
+			return true
+		}
+
+		new, ok := event.ObjectNew.(*goharborv1alpha2.HarborCluster)
+		if !ok {
+			return true
+		}
+		// when status was not changed and spec was not changed, not need reconcile
+		if equality.Semantic.DeepEqual(old.Spec, new.Spec) && old.Status.Status == new.Status.Status {
```
Comment (steven-zou) [inline controllers/goharbor/harborcluster/ctrl_setup.go:112]: It seems `equality.Semantic.DeepDerivative` is better than ` equality.Semantic.DeepEqua`?

**Review by steven-zou — COMMENTED** [review]
see my comments

File: controllers/goharbor/harborcluster/ctrl_setup.go (hunk 1 above)
Comment (chlins) [reply controllers/goharbor/harborcluster/ctrl_setup.go]: updated

**Review by steven-zou — APPROVED** [review]
LGTM

**Review by bitsf — APPROVED** [review]

**Pull request merged**